docs/ml-rfc.md

```diff
@@ -0,0 +1,106 @@
+
+The e2e workflow will be:
+
+1. Kibana sends request to OpenSearch with PPL query, e.g. 
+   ```
+   source log1 | filter time, cpu_load | forecasting forcast_k=10,conf_interval=95
+   ```
+2. The SQL plugin handles this request: to retrieve the index data and do the filtering, and then use the Forecasting PPL command to get the output
+3. The forecasting PPL command will call ML forecasting interface to run the ML algorithm and generate the results. Here is one example of Forecasting algorithm interface:
+    ```
+    input_data: {
+        time: e.g. "2020-09-24 09:10"
+        field_to_forecast: e.g.the value of the field for forecasting. in the above example, it is cpu_load
+    }
+    
+    parameter: {
+       forecast_k: e.g.how many points into the future should be forecasted
+       conf_interval: <1..99> parameter is the confidence interval in percentage around forecasted values. Input an integer between 1 and 99 where a larger number means a greater tolerance for forecast uncertainty. 
+    }
+    
+    output:[ {
+            time: e.g. "2020-09-24 09:10", predicted future time points
+            field_forcasted_value: forcasted future value of the field
+        },
+        ...
+    ]
+   ```
+4. Kibana can use the forecasting result to build a dashboard for customers.
+
+## How to use it for new feature development
+
+As mentioned above, new interfaces will be provided to other plugins including both prediction and training. In the opensearch, transport action is the communication mechanism between plugins. Here are the transport action for prediction and training interfaces.
+
+* Predict Transport Action for prediction job request 
+  ```
+  Request: {
+        "algorithm": "ARIMA",  //the name of algorithm
+        "parameters": {"forecasts_en":10, "seasonal"=true}, // parameters of the algorithm, can be null or empty
+        "modelId":null, //the id for trainded model. can be null
+        "inputData": [[1.0, 2, 3.1, true, "v1"],[1.1, 4, 5.2, false, "v2"]] // internal data frame interface
+    }
+    
+    Response: {
+        "taskId": "123", //the id of the job request
+        "status": "SUCCESS", // the job execution status
+        "predictionResult": [[6.0],[7.0]] // internal data frame interface
+    }
+   ```      
+* Training Transport Action to start training job request - Async Interface
+  ```
+  Request: {
+     "algorithm": "ARIMA", //the name of algorithm
+     "parameters": {"forecasts_en":10, "seasonal"=true}, // parameters of the algorithm, can be null or empty
+     "inputData": [[1.0, 2, 3.1, true, "v1"],[1.1, 4, 5.2, false, "v2"]] // internal data frame interface
+    }
+    
+    
+    Response: {
+     "taskId": "123", //the id of the job request
+     "status": "IN_PROGRESS" // the job execution status
+    
+    }
+   ```
+## Request for Comments:
+
+We would like comments and feedback on the proposal for building Machine Learning Framework here. Some specific questions we’re seeking feedback include
```

Comment:
Create one Github issue to gather customer's feedback and add the link in this doc?

Reply:
Added, thanks for the catch.